.github/workflows/pre-main.yaml

```diff
@@ -160,3 +160,53 @@ jobs:
       - name: (if on main and upstream) Push the newly built image to Quay.io
         if: ${{ github.ref == 'refs/heads/main' && github.repository_owner == 'test-network-function' }}
         run: docker push --all-tags ${REGISTRY}/${IMAGE_NAME}
+ 
+  catalog:
+    name: Update catalog
+    needs: [lint, unit-tests, smoke-tests]
+    runs-on: ubuntu-20.04
+    env:
+      SHELL: /bin/bash
+      KUBECONFIG: '/home/runner/.kube/config'
+
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.sha }}
+
+      - name: Install ginkgo
+        run: go get -u github.com/onsi/ginkgo/ginkgo
+      
+      # make the catalog .md file
+      - name: Create catalog .md file
+        id: catalog-md
+        run: |   
+          make build-catalog-md
+      
+      # if there is any changes in the catalog .md file, commit them and schedule a push
+      - name: Commit files
+        id: commit
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "github-actions"
+          git add -f CATALOG.md
+          if [ -z "$(git status --porcelain)" ]; then
```

Comment:
Is it better to use a condition based on if identifiers.go is changed? We do that in partner repo, the push to quay.io action is executed only if the docker file is changed.

Reply:
I would prefer just looking at changes to the Catalog.md file itself because it will catch all scenarios and is very simple. For instance the first time I committed, there was no change in identifers.go but because of a change in sorting, we still needed to commit and push. 
In which file do we check on identifiers.go to make a decision on when to push?

Reply:
https://github.com/test-network-function/cnf-certification-test-partner/blob/main/.github/workflows/cnf-test-partner-image.yaml

Reply:
I get your point. We can leave it as is for now and optimize later. The action stuff is free within a limit. We might get a bill if we introduce too much excessive work.